src/manifest.c

```diff
@@ -1379,3 +1397,36 @@ int enforce_compliant_manifest(struct file **a, struct file **b, int searchsize,
 	}
 	return ret; // If collisions were found, so manifest is purely additive
 }
+
+/* The function splits a list of manifests into two lists:
+ * - full_manifests: this list will contain all the full manifests that were in the manifests list
+ * - iterative_manifests: this list will contain all the valid iterative manifests from the manifests
+ *   list, plus the full manifest of those bundles that didn't have a valid iterative manifest
+ *   A valid iterative manifest is one that can be used during an update, meaning that the iterative
+ *   manifest must match the "from version" provided, and that it belongs to an already installed bundle*/
+void filter_iterative_manifests(struct list *manifests, int version, struct list **full_manifests, struct list **iterative_manifests)
+{
+	/* Split the manifests into "iterative manifests" and "full manifests" */
+	struct list *list;
+	struct file *file;
+
+	for (list = manifests; list; list = list->next) {
+		file = list->data;
+		/* Keep only iterative manifests that are from the specified version and that are
+		 * already installed */
+		if (file->is_iterative == 1 && file->from_version == version  && is_tracked_bundle(file->bundlename)) {
+			*iterative_manifests = list_prepend_data(*iterative_manifests, file);
+		} else if (file->is_iterative != 1) {
+			*full_manifests = list_prepend_data(*full_manifests, file);
+		}
+	}
+
+	/* Add the full manifest of those bundles that do not have a valid iterative manifest
+	* into the list of iterative_manifests */
+	for (list = *full_manifests; list; list = list->next) {
+		file = list->data;
+		if (!search_bundle_in_manifests(*iterative_manifests, file->bundlename)) {
+			*iterative_manifests = list_prepend_data(*iterative_manifests, file);
```

Comment:
When updating, changed bundles will have an iterative manifest, but unchanged bundles will not. As a result, unchanged bundles will be added to the full_manifests list, downloaded, and parsed.

Reply:
Can we safely skip over unchanged bundles and only download updated ones with iterative manifests?

Reply:
Good question. If a bundle is not changed in the update we should not be attempting to download any iterative or full manifest for that bundle.

Reply:
As jwakre correctly pointed out, the full manifest for all tracked bundles that had not changed is also being downloaded and parsed. I have made some changes so this doesn't happen again. The changes will be updated in a few minutes.